p2p/node/api.go

```diff
@@ -192,7 +190,7 @@ func (p *P2PNode) Request(location common.Location, requestData interface{}, res
 		}
 	}
 
-	go p.requestFromPeers(location, requestData, responseDataType, resultChan)
+	p.requestFromPeers(location, requestData, responseDataType, resultChan)
```

Comment:
This function is only called in a goroutine, so it would still be asynchronous. You may either have `p.requestFromPeers` called in a goroutine or `Request` called in a goroutine, but not both.